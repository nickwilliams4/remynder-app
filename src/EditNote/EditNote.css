button {
  height: 2px;
  cursor: pointer;
}

input {
  border-radius: 10px;
}

.cancel, .save {
  height: 40px;
  width: 120px;
  border-radius: 20px;
  font-size: 16px;
  background-color: black;
  color: white;
}

.button-icon {
  margin-left: 10px;
  zoom: 1.5;
}
.user-name-field {
  height: 30px;
  width: 200px;
}

.cancel:hover, .save:hover {
  color: black;
  background-color: white;
}

.EditNote {
  padding-left: 5px;
  background-color: yellow;
  display: flex;
  justify-content: center;
  align-items: center;
}


@media screen and (min-width: 500px) {
  .EditNote {
    font-size: 25px;
  }
  input[type="text"] {
    font-size: 15px;
  }
  textarea[type="text"] {
    font-size: 20px;
  }
}