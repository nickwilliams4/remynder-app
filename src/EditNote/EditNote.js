import React, { Component } from 'react'
import ApiContext from '../ApiContext'
import config from '../config'
import NotFoundPage from '../NotFoundPage/NotFoundPage'
import { Link } from 'react-router-dom'
import TokenService from '../services/token-service'
import './EditNote.css'

const Required = () => (
  <span className='EditNote__required'>*</span>
)

export default class EditNote extends Component {
  static contextType = ApiContext

  state = {
    error: null,
    id: null,
    title: null,
    content: null,
    created: null,
    remynder: null,
  };

  componentDidMount() {
    fetch(`${config.API_ENDPOINT}/notes`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(error => Promise.reject(error))

        return res.json()
      })
      .then(responseData => {
        this.setState({
          id: responseData.id,
          title: responseData.title,
          content: responseData.content,
          created: responseData.created,
          remynder: responseData.remynder
        })
      })
      .catch(error => {
        this.setState({ error })
      })
  }

  handleChangeTitle = e => {
    this.setState({ title: e.target.value })
  };

  handleChangeContent = e => {
    this.setState({ content: e.target.value })
  };

  handleChangeRemynder = e => {
    this.setState({ remynder: e.target.value })
  };

  handleSubmit = e => {
    e.preventDefault()
    const { note_id } = this.props.match.params
    const { id, title, content, created, remynder } = this.state
    const editNote = { id, title, content, created, remynder}
    fetch(`${config.API_ENDPOINT}/notes/${note_id}`, {
      method: 'PATCH',
      body: JSON.stringify(editNote),
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(error => Promise.reject(error))
      })
      .then(() => {
        this.resetFields(editNote)
        this.context.editNote(editNote)
        this.props.history.push('/NotePage')
      })
      .catch(error => {
        console.error(error)
        this.setState({ error })
      })
  }

  resetFields = (newFields) => {
    this.setState({
      id: newFields.id || '',
      title: newFields.title || '',
      content: newFields.content || '',
      created: newFields.created || '',
      remynder: newFields.remynder || '',
    })
  }

  handleClickCancel = () => {
    this.props.history.push('/')
  };

  render() {
    const { error, title, content } = this.state
    return (
      <NotFoundPage>
        <section className='EditNote'>
          <main role="main">
            <header>
              <h1>Edit Remynder</h1>
            </header>
            <form
              className="EditNote_form"
              onSubmit={this.handleSubmit}
            >
              <div className='EditNote__error' role='alert'>
                {error && <p>{error.message}</p>}
              </div>
              <input
                type='hidden'
                name='id'
              />
              <div className="form-section">
                <label htmlFor="title">
                  Name of Remynder:
                {' '}
                </label>
                <Required />
                <input
                  type="text"
                  title="title"
                  id="title"
                  value={title}
                  required
                  onChange={this.handleChangeTitle}>
                </input>
              </div>
              <div>
                <label htmlFor="content">
                  What's on your mynd?
                {''}
                  <Required />
                </label>
                <textarea
                  title="content"
                  id="content"
                  rows="15"
                  value={content}
                  required
                  onChange={this.handleChangeContent}>
                 </textarea>
              </div>
              <div className="frequency-container">
                <label htmlFor="frequency">How often do you want email remynders? </label>
                <select title="remynder" id="remynder" onChange={this.handleChangeRemynder} >
                <option value="6">Every 6 Hours</option>
                  <option value="12">Every 12 Hours</option>
                  <option value="24">Every 24 Hours</option>
                  <option value="48">Every 2 Days</option>
                </select>
              </div>

              <div className='AddNote__buttons'>
                <Link to='/NotePage'>
                <button type='button' onClick={this.handleClickCancel}>
                  Cancel
          </button>
          </Link>
                {' '}
                <button type='submit'>
                  Save
          </button>
              </div>
            </form>
          </main>
        </section>
      </NotFoundPage>
    )
  }
}