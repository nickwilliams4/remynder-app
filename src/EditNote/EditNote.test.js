import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EditNote from './EditNote'

describe('EditNote component', () => {
  const props = {
    className: 'test-class-name'
  }

  it('renders the EditNote given props', () => {
    const wrapper = shallow(<EditNote {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a form.EditNote by default', () => {
    const wrapper = shallow(<EditNote />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})