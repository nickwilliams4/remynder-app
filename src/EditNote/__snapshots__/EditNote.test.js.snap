// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNote component renders a form.EditNote by default 1`] = `
<NotFoundPage>
  <section
    className="EditNote"
  >
    <main
      role="main"
    >
      <header>
        <h1>
          Edit Remynder
        </h1>
      </header>
      <form
        className="EditNote_form"
        onSubmit={[Function]}
      >
        <div
          className="EditNote__error"
          role="alert"
        />
        <input
          name="id"
          type="hidden"
        />
        <div
          className="form-section"
        >
          <label
            htmlFor="title"
          >
            Name of Remynder:
             
          </label>
          <Required />
          <input
            id="title"
            onChange={[Function]}
            required={true}
            title="title"
            type="text"
            value={null}
          />
        </div>
        <div>
          <label
            htmlFor="content"
          >
            What's on your mynd?
            <Required />
          </label>
          <textarea
            id="content"
            onChange={[Function]}
            required={true}
            rows="15"
            title="content"
            value={null}
          />
        </div>
        <div
          className="frequency-container"
        >
          <label
            htmlFor="frequency"
          >
            How often do you want email remynders? 
          </label>
          <select
            id="remynder"
            onChange={[Function]}
            title="remynder"
          >
            <option
              value="6"
            >
              Every 6 Hours
            </option>
            <option
              value="12"
            >
              Every 12 Hours
            </option>
            <option
              value="24"
            >
              Every 24 Hours
            </option>
            <option
              value="48"
            >
              Every 2 Days
            </option>
          </select>
        </div>
        <div
          className="AddNote__buttons"
        >
          <Link
            to="/NotePage"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </Link>
           
          <button
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </main>
  </section>
</NotFoundPage>
`;

exports[`EditNote component renders the EditNote given props 1`] = `
<NotFoundPage>
  <section
    className="EditNote"
  >
    <main
      role="main"
    >
      <header>
        <h1>
          Edit Remynder
        </h1>
      </header>
      <form
        className="EditNote_form"
        onSubmit={[Function]}
      >
        <div
          className="EditNote__error"
          role="alert"
        />
        <input
          name="id"
          type="hidden"
        />
        <div
          className="form-section"
        >
          <label
            htmlFor="title"
          >
            Name of Remynder:
             
          </label>
          <Required />
          <input
            id="title"
            onChange={[Function]}
            required={true}
            title="title"
            type="text"
            value={null}
          />
        </div>
        <div>
          <label
            htmlFor="content"
          >
            What's on your mynd?
            <Required />
          </label>
          <textarea
            id="content"
            onChange={[Function]}
            required={true}
            rows="15"
            title="content"
            value={null}
          />
        </div>
        <div
          className="frequency-container"
        >
          <label
            htmlFor="frequency"
          >
            How often do you want email remynders? 
          </label>
          <select
            id="remynder"
            onChange={[Function]}
            title="remynder"
          >
            <option
              value="6"
            >
              Every 6 Hours
            </option>
            <option
              value="12"
            >
              Every 12 Hours
            </option>
            <option
              value="24"
            >
              Every 24 Hours
            </option>
            <option
              value="48"
            >
              Every 2 Days
            </option>
          </select>
        </div>
        <div
          className="AddNote__buttons"
        >
          <Link
            to="/NotePage"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </Link>
           
          <button
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </main>
  </section>
</NotFoundPage>
`;
